### PR TITLE
Add SSID scan and validation

### PR DESCRIPTION
**Pull request by brentru**
Resolves https://github.com/adafruit/Adafruit_Wippersnapper_Arduino/issues/328 by adding a function to perform a WiFi scan and check if the user-defined SSID within secrets.json is within the list of scanned SSIDs.

### CODE REVIEW

File: src/network_interfaces/Wippersnapper_ESP8266.h [hunk 1]
```diff
@@ -105,6 +105,48 @@ class Wippersnapper_ESP8266 : public Wippersnapper {
...
+      WS_DEBUG_PRINTLN("ERROR: No WiFi networks found!");
+      return false;
+    }
+
+    // Was the network within secrets.json found?
+    bool is_ssid_found = false;
+    for (int i = 0; i < n; ++i) {
+      if (strcmp(_ssid, WiFi.SSID(i).c_str()) == 0)
+        is_ssid_found = true;
+    }
```
Comment (makermelissa) [inline src/network_interfaces/Wippersnapper_ESP8266.h:133]: Same thing here.

**Review by makermelissa — CHANGES_REQUESTED** [review]
Looks mostly good. I made some comments, but it looks like a lot of similar code. Would having the code in a base class function and then maybe checking an overridable variable to see if the class has been loaded make it easier to maintain?

**Comment by brentru** [comment]
@makermelissa 

> Would having the code in a base class function and then maybe checking an overridable variable to see if the class has been loaded make it easier to maintain?

I'm not sure because each network interface includes its own board-specific implementation/API of the WiFi class

File: src/network_interfaces/Wippersnapper_AIRLIFT.h [hunk 2]
```diff
@@ -95,6 +95,46 @@ class Wippersnapper_AIRLIFT : public Wippersnapper {
...
+      WS_DEBUG_PRINTLN("ERROR: No WiFi networks found!");
+      return false;
+    }
+
+    // Was the network within secrets.json found?
+    bool is_ssid_found = false;
+    for (int i = 0; i < n; ++i) {
+      if (strcmp(_ssid, WiFi.SSID(i)) == 0)
+        is_ssid_found = true;
+    }
```
[the remark this answers is not in the document]
Comment (brentru) [reply src/network_interfaces/Wippersnapper_AIRLIFT.h]: It would! Implemented in next commit.

**Comment by makermelissa** [comment]
> @makermelissa
> 
> > Would having the code in a base class function and then maybe checking an overridable variable to see if the class has been loaded make it easier to maintain?
> 
> I'm not sure because each network interface includes its own board-specific implementation/API of the WiFi class

Fair enough.

**Review by makermelissa — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**